learning_cp/eternity2/features1.jl

```diff
@@ -0,0 +1,198 @@
+#this file contains a few attempts to featurize the Eternity2 problem for learning using FullFeaturedCPNN
+
+struct EternityFeaturization <: SeaPearl.AbstractFeaturization end
+struct EternityFeaturization2 <: SeaPearl.AbstractFeaturization end
+struct EternityFeaturization3 <: SeaPearl.AbstractFeaturization end
+
+function SeaPearl.featurize(sr::SeaPearl.DefaultStateRepresentation{EternityFeaturization,TS}) where TS
+    n = eternity2_generator.n
+    m = eternity2_generator.m
+    k = eternity2_generator.k
+    g = sr.cplayergraph
+    features = zeros(Float32, 17, nv(g))
+    for i in 1:nv(g)
+        cp_vertex = SeaPearl.cpVertexFromIndex(g, i)
+        if isa(cp_vertex, SeaPearl.VariableVertex) && !isa(cp_vertex.variable, SeaPearl.IntVarViewOffset)
+            id = cp_vertex.variable.id
+            #println(id)
+            s = split(id,['_'])[1]
+            l = s[1]
+            t = split(id,['_'])[2]
```

Comment:
this is not very robust to changes on CP core or on the eternity2 generator, but I am not sure we could have done this differently.